File: server/lib/hyperarchy/models/vote.rb
class Vote < Monarch::Model::Record
  column :user_id, :key
  column :election_id, :key
  column :created_at, :datetime
  column :updated_at, :datetime

  belongs_to :user
  belongs_to :election

  def updated
    update(:updated_at => Time.now)
  end
end

File: hyperarchy/client/app/models/vote.js
_.constructor("Vote", Model.Record, {
  constructorInitialize: function() {
    this.columns({
      userId: 'key',
      electionId: 'key',
      updatedAt: 'datetime'
    });

    this.belongsTo('election');
    this.belongsTo('user');
    this.hasMany('rankings');
  },

  formattedUpdatedAt: function() {
    return $.PHPDate("M j, Y @ g:ia", this.updatedAt());
  }
});
